tests/test_database.py

```diff
@@ -0,0 +1,228 @@
+        cluster_username='testuser',
+        cluster_ssh_key='ssh_key'
+    )
+    db_session.add(cluster)
+    db_session.commit()
+
+    # retrieve the cluster from the database and check its fields
+    cluster = ClusterModel.query.filter_by(user_id=user.id).first()
+    assert cluster is not None
+    assert cluster.cluster_name == 'testcluster'
+    assert cluster.cluster_username == 'testuser'
+    assert cluster.cluster_ssh_key == 'ssh_key'
+
+
+def test_simulation_model_creation(db_session: scoped_session):
+    """Test simulation model creation"""
+    # create a new user
+    user = UserModel(username='testuser')
+    user.set_password("testpassword")
+    db_session.add(user)
+    db_session.commit()
+
+    # create a new simulation for the user
+    simulation = SimulationModel(
+        job_id='testjob',
+        user_id=user.id,
+        platform=SimulationModel.Platform.DIRECT.value,
+        input_type=SimulationModel.InputType.YAPTIDE_PROJECT.value,
+        sim_type=SimulationModel.SimType.SHIELDHIT.value,
+        title='testtitle',
+        update_key_hash='testkey'
+    )
+    db_session.add(simulation)
+    db_session.commit()
+
+    # retrieve the simulation from the database and check its fields
+    simulation: SimulationModel = SimulationModel.query.filter_by(user_id=user.id).first()
+    assert simulation.id is not None
+    assert simulation.job_id == 'testjob'
+    assert simulation.platform == SimulationModel.Platform.DIRECT.value
+    assert simulation.input_type == SimulationModel.InputType.YAPTIDE_PROJECT.value
+    assert simulation.sim_type == SimulationModel.SimType.SHIELDHIT.value
+    assert simulation.job_state == SimulationModel.JobState.PENDING.value
+
+
+def test_task_model_creation_and_update(db_session: scoped_session):
+    """Test task model creation"""
+    # create a new user
+    user = UserModel(username='testuser')
+    user.set_password("testpassword")
+    db_session.add(user)
+    db_session.commit()
+
+    # create a new simulation for the user
+    simulation = SimulationModel(
+        job_id='testjob',
+        user_id=user.id,
+        platform=SimulationModel.Platform.DIRECT.value,
+        input_type=SimulationModel.InputType.YAPTIDE_PROJECT.value,
+        sim_type=SimulationModel.SimType.SHIELDHIT.value,
+        title='testtitle',
+        update_key_hash='testkey'
+    )
+    db_session.add(simulation)
+    db_session.commit()
+
+    # create a new task for the simulation
+    task = TaskModel(
+        simulation_id=simulation.id,
+        task_id='testtask',
+        requested_primaries=1000,
+        simulated_primaries=0
+    )
+    db_session.add(task)
+    db_session.commit()
+
+    # retrieve the task from the database and check its fields
+    task: TaskModel = TaskModel.query.filter_by(simulation_id=simulation.id).first()
+    assert task.id is not None
+    assert task.task_state == SimulationModel.JobState.PENDING.value
+
+    sleep(2)
```

Comment:
```suggestion
```

I think that on DB related tests sleeping may not be so useful, it would have more sense on celery tests